### PR TITLE
[location] Add and update doc comments

### PR DESCRIPTION
**Pull request by FelipeACP**
# Why

There were missing doc comments that may display relevant information in the IDE's hover-docs.
Also related to #12239.

# How

I copied all the information from the expo documentation and added it to the source code as documentation comments.

# Test Plan

Preview:
https://i.imgur.com/C3sepg4.png
https://i.imgur.com/snIUOyV.png

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).
- [x] add package mappings to the GenerateDocsAPIData.ts in tools directory
- [x] setup Expo tools and run et gdad or et gdad -p PACKAGE_NAME_IN_MAPPINGS anywhere from the workspace
- [x] verify that new JSON file has been generated in docs/public directory
- [x] import and use APISection component on the unversioned MD for the package
- [x] verify that there are no regression/missing information in comparison to hand-written guide and all types are exported and have the correct types

### CODE REVIEW

**Comment by FelipeACP** [comment]
I've just taken notice of @Simek awesome work on the auto-generation doc feature, so I might as well update the whole package's doc comments to be compatible with his feature.

**Comment by Simek** [comment]
> I've just taken notice of @Simek awesome work on the auto-generation doc feature, so I might as well update the whole package's doc comments to be compatible with his feature.

Hi @FelipeACP, currently the docs auto-gen feature is still in development phase and there is no public guide yet how to add it to the package. I will try to condense the information and will try to list all the required steps below:
* add package mappings to the `GenerateDocsAPIData.ts` in `tools` directory
* setup Expo tools and run `et gdad` or `et gdad -p PACKAGE_|NAME_IN_MAPPINGS` anywhere from the workspace
* verify that new `JSON` file has been generated in `docs/public` directory
* import and use `APISection` component on the `unversioned` MD for the package
* verify that there are no regression/missing information in comparision to hand-written guide and all types are exported and have the correct types

Hope those few steps will help you to achieve the goal described above. Feel free to ask any related questions, if you would have any. 🙂

**Comment by FelipeACP** [comment]
> > I've just taken notice of @Simek awesome work on the auto-generation doc feature, so I might as well update the whole package's doc comments to be compatible with his feature.
> 
> Hi @FelipeACP, currently the docs auto-gen feature is still in development phase and there is no public guide yet how to add it to the package. I will try to condense the information and will try to list all the required steps below:
> 
> * add package mappings to the `GenerateDocsAPIData.ts` in `tools` directory
> * setup Expo tools and run `et gdad` or `et gdad -p PACKAGE_NAME_IN_MAPPINGS` anywhere from the workspace
> * verify that new `JSON` file has been generated in `docs/public` directory
> * import and use `APISection` component on the `unversioned` MD for the package
> * verify that there are no regression/missing information in comparison to hand-written guide and all types are exported and have the correct types
> 
> Hope those few steps will help you to achieve the goal described above. Feel free to ask any related questions, if you would have any. 🙂

I started working on that and I'm having trouble enforcing ordering for Enum values:

image

I'm not sure if I should edit the JSON file generated to fix the order given that it will be overwritten on the next build. 

Aside from that, I believe I just need to remove duplicate types and it's done.

**Comment by Simek** [comment]
> I started working on that and I'm having trouble enforcing ordering for Enum values.

Currently Enums display is unified no matter if Enum has assigned number to the value or not, and values are sorted in alphabetical order. I will work on the improvement like that in the future, I think it's fine for now.

> I'm not sure if I should edit the JSON file generated to fix the order given that it will be overwritten on the next build.

To fix that you will need to alter `APISectionEnums` component, data is generated by TypeDoc, and the one of the goals is to avoid modifying generated JSON, so the update of internals in the future will be possible and quite easy.

**Comment by Simek** [comment]
@FelipeACP the enum ordering by value feature has been merged in #13772 PR, so if you rebase your branch the enum issue should be fixed. 🙂 

Also please check if all the doc comments were up-to-date with the docs descriptions. Usually we we migrate a package if the comments are present in the code they need to be updated.

**Review by tsapeta — APPROVED** [review]
Looks perfect to me, thank you so much for your contribution 🙇 
@Simek, let me know if you have any more suggestions, otherwise I'll merge it 🙂

**Comment by Simek** [comment]
Oh, I thought we have this package conversion already merged in, sorry about that @FelipeACP! 🙏 

Can you rebase and update your PR? It would be nice to merge it in before the SDK 43 release. 🙂

**Comment by Simek** [comment]
Fixes ENG-1961

**Comment by FelipeACP** [comment]
> Oh, I thought we have this package conversion already merged in, sorry about that @FelipeACP! 🙏
> 
> Can you rebase and update your PR? It would be nice to merge it in before the SDK 43 release. 🙂

Sure, I'll resolve the conflicts this weekend and then rebase it.

**Comment by Simek** [comment]
@FelipeACP Hmm, it looks like there is problem with CI after rebase. Did you check out the git submodules and updated them too? If yes, this might be related to the issue in locks.

It might be easier to just close this PR and recreate new one based on the current `master`, then attempting to resolve the CI issues.

File: packages/expo-location/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ### 🛠 Breaking changes
 
-- Dropped support for iOS 11.0 ([#14383](https://github.com/expo/expo/pull/14383) by [@cruzach](https://github.com/cruzach))
```
Comment (Simek) [inline packages/expo-location/CHANGELOG.md:7]: You have removed the changelog entry in here.

**Comment by Simek** [comment]
The CI issues and conflict are related to `react-native` git submodule. Can you reset the state of this file?

**Comment by FelipeACP** [comment]
> The CI issues and conflict are related to `react-native` git submodule. Can you reset the state of this reference?

Sure, I'll get to it today

**Comment by Simek** [comment]
Hi @FelipeACP, as I have described in #14672, I have tried to build upon this PR but it was easier to work on the conversion from the scratch. 🙁 I'm responsible for coordinating the docs conversion progress, and deadline on that module conversion will approach rapidly, so I have decided to dive into this, since there were no response from you for a while. Sorry, hope you can understand that!

**Pull request closed without merging**